reconcile/utils/terraform_client.py

```diff
@@ -553,22 +553,26 @@ def check_output(
         name: str,
         cmd: str,
         return_code: int,
-        stdout: list[str],
-        stderr: list[str],
+        stdout: str,
+        stderr: str,
     ) -> bool:
-        error_occured = False
+        error_occured = return_code != 0
         line_format = "[{} - {}] {}"
         stdout, stderr = self.split_to_lines(stdout, stderr)
         with self._log_lock:
             for line in stdout:
-                logging.debug(line_format.format(name, cmd, line))
-            if return_code == 0:
-                for line in stderr:
+                if line.startswith("[WARN]"):
                     logging.warning(line_format.format(name, cmd, line))
-            else:
+                elif line.startswith("[ERROR]"):
+                    logging.error(line_format.format(name, cmd, line))
+                else:
+                    logging.debug(line_format.format(name, cmd, line))
+            if error_occured:
```

Comment:
We're on Python 3.11 now. You may want to use structural pattern matching (`match ... case`) here?

Reply:
Great! Like this? Just checked all warn and error logs from https://github.com/hashicorp/terraform-provider-aws, there always be a space after `[WARN]` or `[ERROR]`.

```python
                match line.split():
                    case ["[WARN]", *_]:
                        logging.warning(line_format.format(name, cmd, line))
                    case ["[ERROR]", *_]:
                        logging.error(line_format.format(name, cmd, line))
                    case _:
                        logging.debug(line_format.format(name, cmd, line))
```

Reply:
Yeah, exactly! Thx